Goal: Always put the "ro|rw" option in mtab

Fixes: #140397?

Status wrt upstream: Forwarded on 2007/05/30

Author: Steve Langasek <vorlon@debian.org>

Note: 

Index: samba-3.0.25a/source/client/smbmnt.c
===================================================================
--- samba-3.0.25a.orig/source/client/smbmnt.c	2007-05-26 07:45:39.408213553 +0200
+++ samba-3.0.25a/source/client/smbmnt.c	2007-05-26 07:46:33.492644419 +0200
@@ -276,7 +276,10 @@
         ment.mnt_fsname = share_name ? share_name : "none";
         ment.mnt_dir = mount_point;
         ment.mnt_type = "smbfs";
-        ment.mnt_opts = "";
+	if (mount_ro)
+                ment.mnt_opts = "ro";
+        else
+                ment.mnt_opts = "rw";
         ment.mnt_freq = 0;
         ment.mnt_passno= 0;
 
